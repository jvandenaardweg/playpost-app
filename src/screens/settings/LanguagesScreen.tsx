import React from 'react';
import { NavigationScreenProp, NavigationRoute, NavigationStackScreenOptions } from 'react-navigation';
import { ScrollView } from 'react-native';

import { ButtonUpgrade } from '../../components/Header/ButtonUpgrade';
import { LanguagesSelect } from '../../components/LanguagesSelect';

interface Props {
  navigation: NavigationScreenProp<NavigationRoute>;
}

export class SettingsLanguagesScreen extends React.PureComponent<Props> {
  static navigationOptions = ({ navigation }: { navigation: NavigationScreenProp<NavigationRoute> }): NavigationStackScreenOptions => {
    return {
      title: 'Languages',
      headerRight: <ButtonUpgrade onPress={navigation.getParam('handleOnPressUpgrade')} />,
    };
  }

  componentDidMount() {
    this.props.navigation.setParams({ handleOnPressUpgrade: this.handleOnPressUpgrade });
  }

  handleOnPressUpgrade = () => this.props.navigation.navigate('Upgrade');

  handleOnSelectLanguage = (languageName: string) => this.props.navigation.navigate('SettingsVoices', { languageName });

  render() {
    return (
      <ScrollView>
        <LanguagesSelect onSelectLanguage={this.handleOnSelectLanguage} />
      </ScrollView>
    );
  }
}
