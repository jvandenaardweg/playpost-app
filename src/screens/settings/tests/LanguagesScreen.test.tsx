import React from 'react';
// tslint:disable-next-line: no-submodule-imports
import * as ShallowRenderer from 'react-test-renderer/shallow';

import { SettingsLanguagesScreen } from '../LanguagesScreen';

describe('SettingsLanguagesScreen', () => {

  describe('rendering', () => {
    let wrapper: any;

    beforeEach(() => {
      const renderer = ShallowRenderer.createRenderer();
      renderer.render(<SettingsLanguagesScreen />);
      wrapper = renderer.getRenderOutput();
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
