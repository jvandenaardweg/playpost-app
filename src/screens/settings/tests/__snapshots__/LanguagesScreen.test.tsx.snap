// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsLanguagesScreen rendering should render correctly 1`] = `
<Memo>
  <Memo>
    <withNavigation(Connect(LanguagesSelectComponent)) />
  </Memo>
</Memo>
`;
